Added first CI/CD test to verify the build

ErnyTech opened the pull request:
This is the first step in creating a CI/CD infrastructure
for DMD's Snap packages.
The CI/CD tests will be performed through
Github Actions (https://github.com/features/actions), the
executive scripts will be placed in .github/workflows while
the respective Dockerfiles in .github/docker

We need to create a workflow for each "family" of distributions,
for example there will have to be an ubuntu.yml workflow and that
will handle every version of Ubuntu we are going to support.

While the Dockerfiles must be different for each version, they
will be placed in subdirectories of .github/docker to separate
the distribution families, for example there must be the
file .github/docker/ubuntu/Xenial.Dockerfile

Unfortunately since snapd depends on systemd you need to install it,
to make systemd work in Docker I replaced /sbin/udevadm with /bin/true
because otherwise it would have given problems inside the container,
this should not be a problem since udev is not needed to test DMD.

Currently the tests will only interest the DMD build phase, I have
only inserted a small test to verify if at least the builded
Snap package is able to install and perform a simple operation
inside Ubuntu Xenial (dmd --version).

The next works will concern the expansion of the runtime tests and
increasing the number of supported distributions with time.

WebDrake commented:
There's a typo in your commit message: should be `Xenial.Dockerfile` (not `Xeniel`).  Other than that, looks reasonable.

Apologies for the delay in getting back to you, BTW.  It's been a very busy patch ... :-\

Commit:
Added a first CI/CD test to verify the build

This is the first step in creating a CI/CD infrastructure
for DMD's Snap packages.
The CI/CD tests will be performed through
Github Actions (https://github.com/features/actions), the
executive scripts will be placed in .github/workflows while
the respective Dockerfiles in .github/docker

We need to create a workflow for each "family" of distributions,
for example there will have to be an ubuntu.yml workflow that
will handle every version of Ubuntu we intend to support.

While the Dockerfiles must be different for each version, they
will be placed in subdirectories of .github/docker to separate
the distribution families, for example there must be the
file .github/docker/ubuntu/Xenial.Dockerfile

Unfortunately since snapd depends on systemd you need to install it,
to make systemd work in Docker I replaced /sbin/udevadm with /bin/true
because otherwise it would have given problems inside the container,
this should not be a problem since udev is not needed to test DMD.

Currently the tests will only interest the DMD build phase, I have
only inserted a small test to verify if at least the builded
Snap package is able to install and perform a simple operation
inside Ubuntu Xenial (dmd --version).

The next works will concern the expansion of the runtime tests and
increasing the number of supported distributions with time.
